src/main/java/uk/gov/pay/ledger/expungeorredact/service/ExpungeOrRedactService.java

```diff
@@ -118,27 +119,24 @@ private void redactPIIFromTransactionsAndDeleteRelatedEvents() {
 
         noOfTransactionsRedactedMetric.observe(noOfTxsProcessed);
         noOfEventsDeletedRedactedMetric.observe(noOfEventsDeleted);
-
-        if (noOfTxsProcessed > 0) {
-            transactionRedactionInfoDao.update(createdDateOfLastProcessedTransaction);
-        }
     }
 
     private ZonedDateTime getRedactTransactionsUpToDate() {
         return clock.instant()
-                .minus(expungeOrRedactHistoricalDataConfig.getExpungeOrRedactDataOlderThanDays(), ChronoUnit.DAYS)
+                .minus(expungeOrRedactHistoricalDataConfig.getExpungeOrRedactDataOlderThanDays(), DAYS)
                 .atZone(UTC);
     }
 
     private ZonedDateTime getCreatedDateOfLastProcessedTransaction() {
         return ofNullable(transactionRedactionInfoDao.getCreatedDateOfLastProcessedTransaction())
-                .orElseGet(this::getCreatedDateOfFirstTransaction)
+                .orElseGet(this::getDateBeforeCreatedDateOfFirstTransaction)
                 .withZoneSameInstant(UTC);
     }
 
-    private ZonedDateTime getCreatedDateOfFirstTransaction() {
+    private ZonedDateTime getDateBeforeCreatedDateOfFirstTransaction() {
         ZonedDateTime dateOfFirstTransaction = transactionDao.getCreatedDateOfFirstTransaction()
-                .orElseGet(this::getRedactTransactionsUpToDate);
+                .orElseGet(this::getRedactTransactionsUpToDate)
```

Comment:
I think calling `getRedactTransactionsUpToDate` in the `getDayBeforeCreatedDateOfFirstTransaction` method seems a bit confusing. Perhaps this call should be in the `getCreatedDateOfLastProcessedTransaction` method instead?

Reply:
It is in the `getCreatedDateOfLastProcessedTransaction` as `transactionRedactionInfo` will need to be updated with the info created date of the first transaction. If this is not available (local dev env) it will be the date returned by `getRedactTransactionsUpToDate`. So for the first run date range will be  `getRedactTransactionsUpToDate()-1 to getRedactTransactionsUpToDate()`